tasks/run_kops.yml

```
@@ -0,0 +1,178 @@
+      {% endif %}
+      # Don't fail to show that remote has nothing yet
+      kops get secret admin --type SSHPublicKey -o table \
+        --state s3://{{ cluster.s3_bucket_name }} \
+        --name {{ cluster.name }} \
+        | grep -Eo "[:a-f0-9]{16,}" || true
+    source_type: command
+    target_type: command
+  check_mode: False
+  register: _kops_diff_state_sshkey
+  when:
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+
+###
+### Update State Store
+###
+- name: update state store for cluster spec
+  shell: |
+    {% if kops_aws_profile %}
+    # If using boto_profile, make kops aware of it
+    export AWS_PROFILE="{{ kops_aws_profile }}";
+    {% endif %}
+    kops replace -v 9 --force \
+      --state s3://{{ cluster.s3_bucket_name }} \
+      --name {{ cluster.name }} \
+      -f <(echo "{{ lookup('template', 'cluster.yml.j2') }}")
+  args:
+    executable: bash
+  when:
+    - _kops_diff_state_cluster['changed']
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+- name: update state store for instance groups
+  shell: |
+    {% if kops_aws_profile %}
+    # If using boto_profile, make kops aware of it
+    export AWS_PROFILE="{{ kops_aws_profile }}";
+    {% endif %}
+    kops replace -v 9 --force \
+      --state s3://{{ cluster.s3_bucket_name }} \
+      --name {{ cluster.name }} \
+      -f <(echo "{{ lookup('template', 'instance-groups.yml.j2') }}")
+  args:
+    executable: bash
+  when:
+    - _kops_diff_state_instancegroups['changed']
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+- name: update state store for ssh public key
+  shell: |
+    {% if kops_aws_profile %}
+    # If using boto_profile, make kops aware of it
+    export AWS_PROFILE="{{ kops_aws_profile }}";
+    {% endif %}
+    # Delete first
+    kops delete -v 9 secret \
```

instead of *loudly* ignoring the fail 😄, I suggest we also use the blocks.